src/walle-lib/robot.hpp

```diff
@@ -27,7 +27,7 @@ class Robot
     void run(double dt);
 
     /// retourne la capacity de la batterie.
-    virtual double get_battery_capacity() = 0;
+    virtual double get_battery_capacity() {};
```

Comment:
This lines probably produce a warning message because there is no returned value. This can crash or not depending on the compiler.